torch_geometric/loader/link_neighbor_loader.py

```diff
@@ -345,18 +398,62 @@ def get_edge_label_index(
     assert edge_label_index is not None
     assert isinstance(edge_label_index, (list, tuple))
 
-    if isinstance(edge_label_index[0], str):
-        edge_type = edge_label_index
+    if isinstance(data, HeteroData):
+        if isinstance(edge_label_index[0], str):
+            edge_type = edge_label_index
+            edge_type = data._to_canonical(*edge_type)
+            assert edge_type in data.edge_types
+            return edge_type, data[edge_type].edge_index
+
+        assert len(edge_label_index) == 2
+
+        edge_type, edge_label_index = edge_label_index
         edge_type = data._to_canonical(*edge_type)
-        assert edge_type in data.edge_types
-        return edge_type, data[edge_type].edge_index
 
-    assert len(edge_label_index) == 2
+        if edge_label_index is None:
+            return edge_type, data[edge_type].edge_index
 
-    edge_type, edge_label_index = edge_label_index
-    edge_type = data._to_canonical(*edge_type)
+        return edge_type, edge_label_index
 
-    if edge_label_index is None:
-        return edge_type, data[edge_type].edge_index
+    else:  # Tuple[FeatureStore, GraphStore]
+        _, graph_store = data
+        edge_attrs = graph_store.get_all_edge_attrs()
+        edge_types = [attr.edge_type for attr in edge_attrs]
+
+        def _all_indices(ls, val):
+            return [i for i, _ in enumerate(ls) if ls[i] == val]
+
+        def _get_edge_index(edge_type):
```

Comment:
Why not call `GraphStore.coo` instead? Would make the code cleaner. Otherwise, I feel we should just drop this code path here. Seems overly complicated without real use-case.

Reply:
Yeah, that's what the comment was referring to :) Implemented it now.